server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java

```diff
@@ -681,6 +683,50 @@ public void testTransferMaxSeenAutoIdTimestampOnResync() throws Exception {
         }
     }
 
+    public void testAddNewReplicas() throws Exception {
+        try (ReplicationGroup shards = createGroup(between(0, 1))) {
+            shards.startAll();
+            Thread[] threads = new Thread[between(1, 8)];
+            AtomicBoolean isStopped = new AtomicBoolean();
+            boolean appendOnly = randomBoolean();
+            AtomicInteger docId = new AtomicInteger();
+            for (int i = 0; i < threads.length; i++) {
+                threads[i] = new Thread(() -> {
```

Comment:
Does this reproduce the failure without the "fix"?

Reply:
Yep, this test trips the new assertion https://github.com/elastic/elasticsearch/pull/34474/files#diff-0a8902003e3ef16c5ca38aa269765578R683. But only that assertion since we don't enable the MSU optimization in internal engines.

Reply:
sorry, I missed the fact this this is not on the following engine tests. My comments about recovery tests were w.r.t the followengine

Reply:
Ah, no worry. I'll keep this test and add a new test with the following engine.

Reply:
I added a test with the following engine:https://github.com/elastic/elasticsearch/pull/34474/commits/2dfb511abf14e41bc36d09fb06e6c3c849db3b94.